src/colorant.rs

```diff
@@ -134,7 +134,11 @@ impl Colorant {
             .data()
             .xyz(illuminant, Some(self))
             .set_illuminance(100.0);
-        CieLab::try_from(xyz).unwrap()
+        let xyzn = obs
+            .data()
+            .xyz_from_spectrum(&illuminant.spectrum())
+            .set_illuminance(100.0);
```

Comment:
Some code became quite inefficient with this draft solution. It does the full spectrum matrix multiplication twice, to also get the `xyzn` value. This can probably be solved better in some way. But I did not want to spend time optimizing before we had discussed if this is even the correct approach or not.